### PR TITLE
Always use beatmap ruleset in editor gameplay test mode

### LINKED ISSUE #18084: Editor test play inherits current ruleset instead of beatmap's ruleset
### Type

Game behaviour

### Bug description

When entering the editor, everything seems to be fine with the correct ruleset being fetched and applied. however, upon playtesting the beatmap, instead of the player using the correct ruleset for the beatmap, it chooses the current ruleset you have selected in the toolbar. 

### Screenshots or videos

https://user-images.githubusercontent.com/35349837/166618184-be238772-3a5c-4ca1-a571-305f94756995.mp4

### Version

2022.501.0

### Logs

Not entirely sure this is needed for this scenario.

[runtime.log](https://github.com/ppy/osu/files/8617678/runtime.log)

### PR DESCRIPTION
**Pull request by bdach**
- Closes https://github.com/ppy/osu/issues/18084
- Alternative to / closes https://github.com/ppy/osu/pull/18317

### CODE REVIEW

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (frenzibyte) [inline osu.Game/Screens/Edit/EditorLoader.cs:68]: `Beatmap.Value` can't possibly be a `DummyWorkingBeatmap` in editor, can it? `Editor` creates a new working beatmap if it sees it:

https://github.com/ppy/osu/blob/f8830c6850128456266c82de83273204f8b74ac0/osu.Game/Screens/Edit/Editor.cs#L154-L170

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (frenzibyte) [inline osu.Game/Screens/Edit/EditorLoader.cs:69]: I would've updated the ruleset on `Editor` entrance since the entire screen should never operate on a different ruleset than the beatmap's one, but I think we can have it local to test gameplay until that actually proves to be an issue.

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (bdach) [reply osu.Game/Screens/Edit/EditorLoader.cs:68]: It can and not having this change fails tests. This is `EditorLoader`, not `Editor`.

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (bdach) [reply osu.Game/Screens/Edit/EditorLoader.cs:69]: Again, this is `EditorLoader`. It is pushed *before* the editor. It does not load the player.

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (frenzibyte) [reply osu.Game/Screens/Edit/EditorLoader.cs:68]: Aha, so it's done for tests and not real in-game scenarios.

File: osu.Game/Screens/Edit/EditorLoader.cs
Comment (frenzibyte) [reply osu.Game/Screens/Edit/EditorLoader.cs:69]: Right, have confused it with `EditorPlayerLoader`.

**Review by frenzibyte — APPROVED** [review]
Makes sense to me, will wait for a second approval specifically on the leased bindable handling in `SongSelect`, as I'm slightly hesitant about merging this with that part.

File: osu.Game/Screens/Select/SongSelect.cs
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs:796]: I feel like this will fix one other issue that may have been reported in the past... but maybe that was fixed in some other way already.

**Review by peppy — APPROVED** [review]

**Pull request merged**
